examples/kubectl.gpt

```diff
@@ -0,0 +1,42 @@
+tools: sys.write, kubectl, helm, browser
+
+Do not make parallel function calls. Only call one function at a time.
+
+Perform the following tasks in order:
+
+1. Create a Namespace named vote but do not fail if it already exists
+2. Deploy in the vote namespace the application which yaml specification is available at https://luc.run/vote.yaml
+3. Use a single command to wait for all the Pods in the vote namespace to be ready
+4. Install Traefik ingress controller in kube-system namespace with helm only if it is not already installed in this namespace
+5. Make sure the Traefik Pod is in running status
+6. Wait for the IP address of the traefik Service to be available and save it in the file ./lb-ip.txt
+7. Create the file ./ingress.yaml and make sure it contains the yaml specification of an Ingress resource which exposes the vote-ui Service on vote.LBIP.nip.io and the result-ui Service on result.LBIP.nip.io, first making sure to replace the LBIP placeholders with the content of the file ./lb-ip.txt
+8. Create the Ingress resource specified in ./ingress.yaml
+9. Open a browser on vote.LBIP.nip.io but make sur to replace the LBIP placeholder with the content of lb-ip.txt in this URL first
+
+---
+name: kubectl
```

Comment:
I can't remember if we discussed this, but this is technically a "safer" way to implement:

```
name: kubectl
description: use kubectl command to manage k8s resources
args: arguments: the arguments for the kubectl command

!#/bin/bash

kubectl ${arguments}
```


Can you do it this way or was there a particular reason you didn't? I can't recall the details of the convos we had around this.